<?xml version="1.0"?>

<launch>
    <!-- The PWM Values associated with each input -->
    <arg name="max_roll"        default="1925"/>
    <arg name="min_roll"        default="1103"/>

    <arg name="max_pitch"       default="1925"/>
    <arg name="min_pitch"       default="1103"/>

    <arg name="max_yaw"         default="1925"/>
    <arg name="min_yaw"         default="1111"/>

    <arg name="max_thrust"    default="1925"/>
    <arg name="min_thrust"    default="1105"/>

    <arg name="manual_mode"     default="965"/>
    <arg name="althold_mode"    default="1515"/>
    <arg name="computer_mode"   default="2065"/>

    <arg name="emergency_stop"  default="965"/>
    <arg name="land"            default="2065"/>

    <!-- The normalization ranges for the stick inputs -->
    <arg name="roll_range_max"  default="2.5"/>
    <arg name="roll_range_min"  default="-2.5"/>

    <arg name="pitch_range_max"  default="2.5"/>
    <arg name="pitch_range_min"  default="-2.5"/>

    <arg name="yaw_range_max"  default="2.5"/>
    <arg name="yaw_range_min"  default="-2.5"/>

    <arg name="thrust_range_max"  default="2.5"/>
    <arg name="thrust_range_min"  default="-2.5"/>

    <arg name="namespace"       default="tello"/>

    <!-- RC Teleop node -->
	<group ns="$(arg namespace)">
        <node pkg="tello_driver"       type="rc_teleop_node" name="rc_teleop" output="screen">
            <param name="max_roll"       type="double" value="$(arg max_roll)"/>
            <param name="min_roll"       type="double" value="$(arg min_roll)"/>
            <param name="max_pitch"      type="double" value="$(arg max_pitch)"/>
            <param name="min_pitch"      type="double" value="$(arg min_pitch)"/>
            <param name="max_yaw"        type="double" value="$(arg max_yaw)"/>
            <param name="min_yaw"        type="double" value="$(arg min_yaw)"/>
            <param name="max_thrust"   type="double" value="$(arg max_thrust)"/>
            <param name="min_thrust"   type="double" value="$(arg min_thrust)"/>
            <param name="manual_mode"    type="double" value="$(arg manual_mode)"/>
            <param name="althold_mode"   type="double" value="$(arg althold_mode)"/>
            <param name="computer_mode"  type="double" value="$(arg computer_mode)"/>
            <param name="emergency_stop" type="double" value="$(arg emergency_stop)"/>
            <param name="land"           type="double" value="$(arg land)"/>
            <param name="roll_range_max" type="double" value="$(arg roll_range_max"/>
            <param name="roll_range_min" type="double" value="$(arg roll_range_min"/>
            <param name="pitch_range_max" type="double" value="$(arg pitch_range_max"/>
            <param name="pitch_range_min" type="double" value="$(arg pitch_range_min"/>
            <param name="yaw_range_max" type="double" value="$(arg yaw_range_max"/>
            <param name="yaw_range_min" type="double" value="$(arg yaw_range_min"/>
            <param name="thrust_range_max" type="double" value="$(arg thrust_range_max"/>
            <param name="thrust_range_min" type="double" value="$(arg thrust_range_min"/>
        </node>
    </group>

</launch>